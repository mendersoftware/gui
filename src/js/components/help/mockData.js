// Copyright 2020 Northern.tech AS
//
//    Licensed under the Apache License, Version 2.0 (the "License");
//    you may not use this file except in compliance with the License.
//    You may obtain a copy of the License at
//
//        http://www.apache.org/licenses/LICENSE-2.0
//
//    Unless required by applicable law or agreed to in writing, software
//    distributed under the License is distributed on an "AS IS" BASIS,
//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//    See the License for the specific language governing permissions and
//    limitations under the License.
export const helpProps = {
  docsVersion: undefined,
  isHosted: false,
  menderVersion: 'master',
  isEnterprise: true,
  tenantCapabilities: {
    hasDeviceConfig: true,
    hasMonitor: true
  },
  tokens: [],
  versions: {
    repos: {
      mender: '3.3.0',
      'mender-artifact': '3.8.0',
      'mender-binary-delta': '1.3.1',
      'mender-cli': '1.8.0',
      'mender-configure-module': '1.0.4',
      'mender-connect': '2.0.2',
      'mender-convert': '3.0.0',
      'mender-gateway': '1.0.0',
      'monitor-client': '1.2.0'
    },
    releaseDate: '2003-01-01'
  }
};
