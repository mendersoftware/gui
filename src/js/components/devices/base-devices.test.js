// Copyright 2021 Northern.tech AS
//
//    Licensed under the Apache License, Version 2.0 (the "License");
//    you may not use this file except in compliance with the License.
//    You may obtain a copy of the License at
//
//        http://www.apache.org/licenses/LICENSE-2.0
//
//    Unless required by applicable law or agreed to in writing, software
//    distributed under the License is distributed on an "AS IS" BASIS,
//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//    See the License for the specific language governing permissions and
//    limitations under the License.
import React from 'react';

import { defaultState, undefineds } from '../../../../tests/mockData';
import { render } from '../../../../tests/setupTests';
import {
  AcceptedEmptyState,
  DefaultAttributeRenderer,
  DeviceCreationTime,
  DeviceSoftware,
  DeviceStatusRenderer,
  DeviceTypes,
  PendingEmptyState,
  PreauthorizedEmptyState,
  RejectedEmptyState,
  RelativeDeviceTime,
  defaultTextRender,
  getDeviceIdentityText
} from './base-devices';

describe('smaller components', () => {
  [
    AcceptedEmptyState,
    DefaultAttributeRenderer,
    DeviceCreationTime,
    DeviceSoftware,
    DeviceStatusRenderer,
    DeviceTypes,
    PendingEmptyState,
    PreauthorizedEmptyState,
    RejectedEmptyState,
    RelativeDeviceTime
  ].forEach(Component => {
    it(`renders ${Component.displayName || Component.name} correctly`, () => {
      const { baseElement } = render(
        <Component
          allCount={10}
          device={defaultState.devices.byId.a1}
          filters={[]}
          highlightHelp={true}
          idAttribute={{ attribute: 'mac', scope: 'identity' }}
          column={{ title: 'mac', attribute: { name: 'mac', scope: 'identity' }, sortable: true, textRender: defaultTextRender }}
          limitMaxed={true}
          onClick={jest.fn}
        />
      );
      const view = baseElement.firstChild.firstChild;
      expect(view).toMatchSnapshot();
      expect(view).toEqual(expect.not.stringMatching(undefineds));
    });
  });

  it(`uses getDeviceIdentityText correctly`, () => {
    let result = getDeviceIdentityText({ device: defaultState.devices.byId.a1 });
    expect(result).toMatch(defaultState.devices.byId.a1.id);
    expect(result).toEqual(expect.not.stringMatching(undefineds));
    result = getDeviceIdentityText({ device: defaultState.devices.byId.a1, idAttribute: { attribute: 'mac', scope: 'identity' } });
    expect(result).toMatch(defaultState.devices.byId.a1.identity_data.mac);
    expect(result).toEqual(expect.not.stringMatching(undefineds));
  });
  it(`uses defaultTextRender correctly`, () => {
    let result = defaultTextRender({ column: { title: 'mac', attribute: { name: 'id', scope: 'identity' } }, device: defaultState.devices.byId.a1 });
    expect(result).toMatch(defaultState.devices.byId.a1.id);
    expect(result).toEqual(expect.not.stringMatching(undefineds));
    result = defaultTextRender({ column: { title: 'mac', attribute: { name: 'mac', scope: 'identity' } }, device: defaultState.devices.byId.a1 });
    expect(result).toMatch(defaultState.devices.byId.a1.identity_data.mac);
    expect(result).toEqual(expect.not.stringMatching(undefineds));
  });
});
