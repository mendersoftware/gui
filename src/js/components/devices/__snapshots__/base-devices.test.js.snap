// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smaller components renders AcceptedEmptyState correctly 1`] = `
<div
  class="dashboard-placeholder"
>
  <p>
    No devices found
  </p>
</div>
`;

exports[`smaller components renders DefaultAttributeRenderer correctly 1`] = `
<div
  title="dc:a6:32:12:ad:bf"
>
  <div
    class="text-overflow"
  >
    dc:a6:32:12:ad:bf
  </div>
</div>
`;

exports[`smaller components renders DeviceCreationTime correctly 1`] = `
<div>
  <time
    datetime="2019-01-13T06:25:00+00:00"
  >
    2019-01-13 06:25
  </time>
</div>
`;

exports[`smaller components renders DeviceSoftware correctly 1`] = `
<div
  title="-"
>
  <div
    class="text-overflow"
  >
    -
  </div>
</div>
`;

exports[`smaller components renders DeviceStatusRenderer correctly 1`] = `
<div>
  <div
    class="margin-right-small"
  >
    accepted
  </div>
</div>
`;

exports[`smaller components renders DeviceTypes correctly 1`] = `
<div
  title="raspberrypi4"
>
  <div
    class="text-overflow"
  >
    raspberrypi4
  </div>
</div>
`;

exports[`smaller components renders PendingEmptyState correctly 1`] = `
<div
  class="dashboard-placeholder"
>
  <p>
    There are no devices pending authorization
  </p>
  <p>
    Visit the 
    <a
      href="/help/get-started"
    >
      Help section
    </a>
     to learn how to connect devices to the Mender server.
  </p>
</div>
`;

exports[`smaller components renders PreauthorizedEmptyState correctly 1`] = `
<div
  class="dashboard-placeholder"
>
  <p>
    There are no preauthorized devices.
  </p>
  <img
    alt="preauthorize"
    src="preauthorize.png"
  />
</div>
`;

exports[`smaller components renders RejectedEmptyState correctly 1`] = `
<div
  class="dashboard-placeholder"
>
  <p>
    There are no rejected devices
  </p>
</div>
`;

exports[`smaller components renders RelativeDeviceTime correctly 1`] = `
<div>
  <time
    datetime="2019-01-01"
  >
    12 days ago
  </time>
</div>
`;
